sdk/storage/azure-storage-file-share/azure/storage/fileshare/_policies.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import Any, TYPE_CHECKING
+
+from azure.core.pipeline.policies import (
+    SansIOHTTPPolicy
+)
+from azure.core.exceptions import ServiceRequestError
+from ._shared.request_handlers import throw_exception_if_contains_header
+from ._generated.version import VERSION
+
+if TYPE_CHECKING:
+    from azure.core.pipeline import PipelineRequest
+
+START_COPY_OPERATION = 'start_copy_from_url'
+
+
+class StorageVersionCheckPolicy(SansIOHTTPPolicy):
+    def on_request(self, request):
+        # type: (PipelineRequest, Any) -> None
+        headers = request.http_request.headers
+        request_service_version = headers['x-ms-version']
+
+        if request_service_version == VERSION:
+            return
+
+        # 2019 - 07 - 07 check
+        if request_service_version < '2019-07-07':
+            # File Lease
+            throw_exception_if_contains_header(headers, "x-ms-lease-id", "any file API", request_service_version)
+            throw_exception_if_contains_header(headers, "x-ms-lease-duration", "any file API", request_service_version)
+            throw_exception_if_contains_header(headers, "x-ms-proposed-lease-id", "any file API", request_service_version)
+
+            if "comp=lease" in request.http_request.url:
```

Comment:
is there a way to check in the query string instead of whole URL?

Reply:
yes there is. We can check if query.get('comp') == 'lease'